src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Cvt.kt

```diff
@@ -75,37 +79,47 @@ class Cvt : AbstractCommand(
         val inputSplit = inputPattern.find(input) ?: return
         // Destructuring the list
         val (_, sourceValue, sourceUnit) = inputSplit.groupValues
+        val srcUnitLower = sourceUnit.toLowerCase()
 
-        if (!sourceUnit.isCompatibleWithUnit(targetUnit)) {
+        if (!srcUnitLower.isCompatibleWithUnit(targetUnit)) {
             channel.sendMessage("<:YoichiLOL:701312070880329800> I wish that that was possible as well mate.").queue()
             return
         }
 
-        val sourceDouble = sourceValue.toDouble()
+        val sourceFloat = sourceValue.toFloat()
 
-        val converted = if (lengths.contains(targetUnit)) {
-            TODO("Length conversion")
-        } else {
-            val kelvin = sourceDouble.toKelvin(sourceUnit)
+        val converted = when {
+            lengths.contains(targetUnit) -> convertLength(sourceFloat, sourceUnit, targetUnit)
+            weights.contains(targetUnit) -> {
+                val source = sourceUnit.toMassUnit()
+                val target = targetUnit.toMassUnit()
+                val converter = source.getConverterTo(target)
+                val measure = Measure.valueOf(sourceFloat, source).doubleValue(source)
 
-            if (kelvin < 0 || kelvin > 10F.pow(32F)) {
-                val highLow = if (kelvin < 0) "low" else "high"
-
-                channel.sendMessage("<:HiroOhGod:701312362401103902> Temperatures that $highLow are not possible.")
-                    .queue()
-                return
+                converter.convert(measure).toFloat()
             }
```

Comment:
Maybe check for negative values for both lengths and weights? Can't have a negative weight or distance after all. Not sure if your length converter will throw an error if the value is negative.